src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -40,29 +41,38 @@ def _resource_has_changed(
     db_engine:  sqlalchemy.engine.base.Engine,
     id_column: str = "id"
 ) -> bool:
-    query = f"SELECT * FROM {resource_name} where {id_column} = {resource[id_column]} limit 1;"
+    query = f"SELECT * FROM {resource_name} where SourceId = {resource[id_column]} limit 1;"
+    db_item: dict = {}
     with db_engine.connect() as con:
         result: Union[ResultProxy, None] = con.execute(query)
         if result is None:
             return True
         for row in result:
-            for column, value in row.items():
-                if column != "CreateDate" and column != "LastModifiedDate":
-                    if resource[column] != value:
-                        return True
+            for column, db_value in row.items():
+                db_item = {**db_item, column: db_value}
+    json_response = r"%s" % db_item["JsonResponse"]
+    db_parsed_item = json.loads(json_response)
+
+    for key in db_parsed_item:
+        api_value = resource[key]
+        db_value = db_parsed_item[key]
+
+        if isinstance(api_value, dict) or isinstance(api_value, list):
+            api_value = str(api_value).replace("'", '"')
+            db_value = str(db_value).replace("'", '"')
```

Comment:
Is this conversion still necessary?

Reply:
Yes, for complex types it is, it could be improved, but it is necessary.